re_data_ui/src/components/FlowGraph.tsx

```diff
@@ -0,0 +1,202 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import React, {
+  MouseEvent as ReactMouseEvent, ReactElement, useCallback,
+  useEffect, useRef, useState,
+} from 'react';
+import ReactFlow, {
+  ConnectionLineType, Controls, Edge, Elements, getIncomers, getOutgoers,
+  isEdge, isNode, Node, OnLoadParams, ReactFlowProvider,
+  removeElements,
+} from 'react-flow-renderer';
+import { useSearchParams } from 'react-router-dom';
+import '../graph.css';
+import { getLayoutElements } from '../utils';
+import CustomNode from './CustomNode';
+
+export interface Props {
+  data: Elements;
+  disableClick?: boolean;
+}
+
+const nodeTypes = {
+  'custom-node': CustomNode,
+};
+
+function FlowGraph(params: Props): ReactElement {
+  const { data, disableClick = false } = params;
+  const instanceRef = useRef<OnLoadParams | null>(null);
+  const [, setURLSearchParams] = useSearchParams();
+
+  const res = getLayoutElements(data);
+  const [elements, setElements] = useState<Elements>(res);
+
+  useEffect(() => {
+    setElements(res);
+  }, [data]);
+
+  const onElementsRemove = (elementsToRemove: Elements) => {
+    setElements((els) => removeElements(elementsToRemove, els));
+  };
+
+  const getAllIncomer = (node: Node, el: Elements): Node[] => getIncomers(node, el).reduce(
+    (memo: any, incomer) => [
+      ...memo,
+      incomer,
+      ...getAllIncomer(incomer, elements),
+    ],
+    [],
+  );
+
+  const getAllOutgoer = (node: Node, el: Elements): Node[] => getOutgoers(node, el).reduce(
+    (memo: any, outgoer) => [
+      ...memo,
+      outgoer,
+      ...getAllOutgoer(outgoer, el),
+    ],
+    [],
+  );
```

Comment:
I see this does a reduce operation in a recursive manner, we should discuss this in full detail.

Reply:
funny enough @mateuszklimek  said it's not needed,

the idea is to get all the edges linked to it (its direct/indirect parents and children)

but i'll refactor it to get only its direct children.